core/src/main/java/org/elasticsearch/rest/action/support/RestActions.java

```diff
@@ -93,6 +108,95 @@ public static void buildBroadcastShardsHeader(XContentBuilder builder, ToXConten
         builder.endObject();
     }
 
+    /**
+     * Create the XContent header for any {@link BaseNodesResponse}.
+     *
+     * @param builder XContent builder.
+     * @param params XContent parameters.
+     * @param response The response containing individual, node-level responses.
+     * @see #buildNodesHeader(XContentBuilder, Params, int, int, int, List)
+     */
+    public static <NodeResponse extends BaseNodeResponse> void buildNodesHeader(final XContentBuilder builder, final Params params,
+                                                                                final BaseNodesResponse<NodeResponse> response)
+        throws IOException {
+        final int successful = response.getNodes().size();
```

Comment:
I understand the thing about not all subclasses implementing toXContent and I agree doing it this way is better in that case. If I had to do it myself I'm not sure I'd do it this way but I don't know how I'd do it. In the case that not all subclasses have toXContent then this is fine/good/maybe the best way.